tests/test_data/test_monitor_data.py

```diff
@@ -611,6 +611,75 @@ def isel(data, freqs):
     assert len(dot.f) == 2
 
 
+def test_translated_copy():
+    mode_data = make_mode_solver_data()
+    field_data = make_field_data_2d()
+
+    vector = (1, 0, 0)
+    mode_data_translated = mode_data.translated_copy(vector=vector)
+    field_data_translated = field_data.translated_copy(vector=vector)
+
+    field1 = mode_data.symmetry_expanded_copy.Ex.isel(mode_index=0, f=0)
+    field2 = mode_data_translated.symmetry_expanded_copy.Ex.isel(mode_index=0, f=0)
+
+    # import matplotlib
+    # matplotlib.use('TkAgg')
+    # from matplotlib import pyplot as plt
+    # plt.figure()
+    # plt.imshow(np.squeeze(np.abs(field1.data - field2.data)))
+    # plt.colorbar()
+    # plt.show()
+
+    atol = 1e-10
+
+    assert np.allclose(field1.data, field2.data)
+
+    assert np.allclose(
+        mode_data.dot(mode_data), mode_data_translated.dot(mode_data_translated), atol=atol
+    )
+    assert np.allclose(
+        mode_data.outer_dot(mode_data),
+        mode_data_translated.outer_dot(mode_data_translated),
+        atol=atol,
+    )
+    assert np.allclose(
+        mode_data.dot(field_data), mode_data_translated.dot(field_data_translated), atol=atol
+    )
+    assert np.allclose(
+        mode_data.outer_dot(field_data),
+        mode_data_translated.outer_dot(field_data_translated),
+        atol=atol,
+    )
+    assert np.allclose(
+        field_data.dot(mode_data), field_data_translated.dot(mode_data_translated), atol=atol
+    )
+    assert np.allclose(
+        field_data.outer_dot(mode_data),
+        field_data_translated.outer_dot(mode_data_translated),
+        atol=atol,
+    )
+    assert np.allclose(
+        field_data.dot(field_data), field_data_translated.dot(field_data_translated), atol=atol
+    )
+    assert np.allclose(
+        field_data.outer_dot(field_data),
+        field_data_translated.outer_dot(field_data_translated),
+        atol=atol,
+    )
+
+    assert np.allclose(
+        mode_data.dot(mode_data),
+        mode_data_translated.translated_copy(vector=[-v for v in vector]).dot(mode_data),
+        atol=atol,
+    )
+
+    assert np.allclose(
+        mode_data.outer_dot(mode_data),
+        mode_data_translated.translated_copy(vector=[-v for v in vector]).outer_dot(mode_data),
+        atol=atol,
+    )
```

Comment:
These tests make sure that `.dot` and `.outer_dot` work with translated data, that translating two sets of data by the same amount returns the same overlap, and that translating back recovers the original data. That's great, but is there also a way to test that translation does what we expect? For example, you could do two mode solves for the same waveguide, but the waveguide is translated, and then check that the overlap between the translated data and the second data is close to 1? This could be added to the mode solver tests.